pkg/util/buildkitutil/buildkit.go

```diff
@@ -20,21 +20,33 @@ import (
 )
 
 const buildkitConfigTemplate = `
-[registry."{{ if .Registry }}{{ .Registry }}{{ else }}docker.io{{ end }}"]{{ if .Mirror }}
-  mirrors = ["{{ .Mirror }}"]{{ end }}
-  http = {{ .UseHTTP }}
-  {{ if .SetCA}}ca=["/etc/registry/ca.pem"]
-  [[registry."{{ if .Registry }}{{ .Registry }}{{ else }}docker.io{{ end }}".keypair]]
-	key="/etc/registry/key.pem"
-	cert="/etc/registry/cert.pem"
-  {{ end }}
+[registry]
+{{- range $index, $value := .RegistryName }}
+  [registry."{{ if $value }}{{ $value }}{{ else }}docker.io{{ end }}"]
+    {{- if index $.UseHTTP $index }}
+    http = true
+    {{- end }}
+    {{- if $.Mirror }}
+    mirrors = ["{{ $.Mirror }}"]
+    {{- end }}
+    {{- if index $.CaPath $index }}
+    ca=["/etc/registry/{{ $value }}_ca.pem"]
+    {{- end }}
+    {{- if and (index $.CertPath $index) (index $.KeyPath $index) }}
+    [[registry."{{ if $value }}{{ $value }}{{ else }}docker.io{{ end }}".keypair]]
+      key="/etc/registry/{{ $value }}_key.pem"
+      cert="/etc/registry/{{ $value }}_cert.pem"
+    {{- end }}
+{{- end }}
 `
 
 type BuildkitConfig struct {
-	Registry string
-	Mirror   string
-	UseHTTP  bool
-	SetCA    bool
+	RegistryName []string
+	CaPath       []string
+	CertPath     []string
+	KeyPath      []string
+	UseHTTP      []bool
+	Mirror       string
 }
```

Comment:
Can we use `RegistriesData` directly? Thus we don't need to do the redundant `append` after the JSON parsing. BTW I think the name `BuildkitConfig` is more suitable.

Reply:
Sure that's possible. But I think then we would have to change the JSON format. Something like this:

`{
    "mirror": ["dockerhub-mirror", "dockerhub-mirror"]
    "registry_name": ["myRegistry1", "myRegistry2"],
    "ca_path": ["myCa1", "myCa2"],
    "key_path": ["", "myKey2"],
    "cert_path": ["", "myCert2"],
    "use_http": [false, false]
}`

The format I proposed in the pr is nested, so I don't think unmarshaling directly into BuildkitConfig is possible. But if we change the JSON format to the above, I think it's possible.

What do you think? Personally I think its a tradeoff between user readability and code simplicity. But if we're being practical, I don't think a user will be specifying more than two registries, so I think the decrease in readability is ok.

Reply:
I think we don't have to change the JSON format. The current format used by `RegistriesData` (might need to add the mirror to `Registry`) LGTM. I mean they can be the same struct. No need to maintain two similar structs.

Reply:
To clarify, are you suggesting to move `RegistriesData` and `Registry` to `buildkit.go`? And then we would have to change `buildkitConfigTemplate` right? Because right now `buildkitConfigTemplate` is based on the format of `BuildkitConfig`.

Reply:
Yes.